app/interactors/tasks/destroy.rb

```diff
@@ -2,10 +2,22 @@ module Tasks
   class Destroy
     include Interactor
 
-    delegate :task, to: :context
+    delegate :task, :project, :user, to: :context
+
+    before do
+      @task_name = task.name
+      @task_description = task.description
```

Comment:
please don't use here `@var`, you can get it in mailer if you need it in mailer

```suggestion
      @task_name = task.name
      @task_description = task.description
```